runtime/pkg/metricssql/filter_parser.go

```diff
@@ -0,0 +1,350 @@
+			Operator: op,
+			Expressions: []*metricsview.Expression{
+				expr,
+				{Value: pattern},
+			},
+		},
+	}, nil
+}
+
+func parseBetween(ctx context.Context, n *ast.BetweenExpr, q *query) (*metricsview.Expression, error) {
+	expr, err := parseFilter(ctx, n.Expr, q)
+	if err != nil {
+		return nil, err
+	}
+
+	left, err := parseFilter(ctx, n.Left, q)
+	if err != nil {
+		return nil, err
+	}
+
+	right, err := parseFilter(ctx, n.Right, q)
+	if err != nil {
+		return nil, err
+	}
+	return &metricsview.Expression{
+		Condition: &metricsview.Condition{
+			Operator: metricsview.OperatorAnd,
+			Expressions: []*metricsview.Expression{
+				{
+					Condition: &metricsview.Condition{
+						Operator:    metricsview.OperatorGte,
+						Expressions: []*metricsview.Expression{expr, left},
+					},
+				},
+				{
+					Condition: &metricsview.Condition{
+						Operator:    metricsview.OperatorLte,
+						Expressions: []*metricsview.Expression{expr, right},
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+func parseValueExpr(in ast.Node) (string, error) {
+	node, ok := in.(ast.ValueExpr)
+	if !ok {
+		return "", fmt.Errorf("metrics sql: expected value expression, got %T", in)
+	}
+	var sb strings.Builder
+	rctx := format.NewRestoreCtx(format.RestoreNameBackQuotes|format.RestoreStringWithoutCharset, &sb)
+	if err := node.Restore(rctx); err != nil {
+		return "", err
+	}
+	return sb.String(), nil
+}
+
+func parseTimeUnitValueExpr(in ast.Node) (string, error) {
+	node, ok := in.(*ast.TimeUnitExpr)
+	if !ok {
+		return "", fmt.Errorf("metrics sql: expected time_unit value expression, got %T", in)
+	}
+	return node.Unit.String(), nil
+}
+
+func parseFuncCallInFilter(ctx context.Context, node *ast.FuncCallExpr, q *query) (*metricsview.Expression, error) {
+	switch node.FnName.L {
+	case "time_range_start":
+		if q == nil {
+			return nil, fmt.Errorf("metrics sql: time_range_start function is only supported for metrics_sql")
+		}
+		return q.parseTimeRangeStart(ctx, node)
+	case "time_range_end":
+		if q == nil {
+			return nil, fmt.Errorf("metrics sql: time_range_end function is only supported for metrics_sql")
+		}
+		return q.parseTimeRangeEnd(ctx, node)
+	case "now":
+		return &metricsview.Expression{
+			Value: time.Now().Format(time.RFC3339),
+		}, nil
+	case "date_add", "date_sub": // ex : date_add(time, INTERVAL x UNIT)
+		val, err := parseFilter(ctx, node.Args[0], q) // handling of time
+		if err != nil {
+			return nil, err
+		}
+		timeStr, ok := val.Value.(string)
+		if !ok {
+			return nil, fmt.Errorf("metrics sql: expected time value in date_add/date_sub function")
+		}
+		t, err := dateparse.ParseAny(timeStr)
+		if err != nil {
+			return nil, err
+		}
+
+		expr, err := parseValueExpr(node.Args[1]) // handling of x
+		if err != nil {
+			return nil, err
+		}
+		amt, err := strconv.Atoi(expr)
+		if err != nil {
+			return nil, fmt.Errorf("metrics sql: expected integer value in date_add/date_sub function")
+		}
+
+		// TODO :: check this
```

Comment:
Does it need to be handled before merging?

Reply:
I had actually checked this. Forgot to remove todo :)